### PR TITLE
Event 352

### PR DESCRIPTION
**Pull request by dsgoers**
Removed readyToProcess and prevent users from attempting partial credit card refunds within 24 hours of the payment (this is also enforced by the API).

### CODE REVIEW

File: app/scripts/controllers/paymentModal.js [hunk 1]
```diff
@@ -143,16 +141,45 @@ angular.module('confRegistrationWebApp')
...
         refundedPaymentId: payment.id,
         registrationId: payment.registrationId,
         paymentType: 'REFUND',
-        refundChannel: payment.paymentType,
-        readyToProcess: true
+        refundChannel: payment.paymentType
       };
     };
 
+    $scope.refreshRefund = function (refund) {
```
Comment (adammeyer) [inline app/scripts/controllers/paymentModal.js]: Can you explain what this function is for?

Comment (dsgoers) [inline app/scripts/controllers/paymentModal.js]: It determines whether a partial refund is available, and if not, disables the amount window and shows the warning.

File: app/scripts/controllers/paymentModal.js [hunk 2]
```diff
@@ -143,16 +141,45 @@ angular.module('confRegistrationWebApp')
...
+      var paymentToRefund;
+
+      _.forEach($scope.registration.pastPayments, function(value) {
```
Comment (adammeyer) [inline app/scripts/controllers/paymentModal.js]: You can use the _.find here.

File: app/scripts/controllers/paymentModal.js [hunk 3]
```diff
@@ -311,4 +338,22 @@ angular.module('confRegistrationWebApp')
       var registrationPromoCodes = _.pluck($scope.registration.promotions, 'id');
       return !_.contains(registrationPromoCodes, p.id);
     };
+
+    /*
+    Returns false is payment if payment and refund channel are credit card and current time is less than 24 hours from
+    payment time.  If a refund is issued before the payment settles, authorize.net will do a full refund regardless of amount.
+    */
+    $scope.isPartialRefundAvailable = function(payment, refundChannel) {
```
Comment (adammeyer) [inline app/scripts/controllers/paymentModal.js]: This doesn't need to be a $scope function if it's not being accessed by the view.  Change this to var isPartialRefundAvailable = function...

Comment (adammeyer) [inline app/scripts/controllers/paymentModal.js]: Or, you could keep this and remove $scope.onlyFullRefund.  It looks like that is just being set to the same value as $scope.isPartialRefundAvailable.

Comment (dsgoers) [inline app/scripts/controllers/paymentModal.js]: I added `$scope` to it so I could call it in the tests.

Comment (adammeyer) [inline app/scripts/controllers/paymentModal.js]: Okay.  It should only be called during the digest cycle.  What do you mean by a lot?

**Comment by dsgoers** [comment]
All changes addressed.

**Comment by adammeyer** [comment]
👍

**Pull request merged**